packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -131,11 +135,85 @@ export class TriggerManager extends Component {
       this.handleError(error)
     }
   }
+
+  /**
+   * Get the ID of the cipher selected by the user in the list
+   *
+   * @returns {string} the cipher ID
+   */
+  getSelectedCipherId() {
+    const { selectedCipher } = this.state
+    return selectedCipher && selectedCipher.id
+  }
+
+  /**
+   * Create a new cipher and return its ID
+   *
+   * @param {string} login - the login to register in the new cipher
+   * @param {string} password - the password to register in the new cipher
+   *
+   * @returns {string} the cipher ID
+   */
+  async getNewCipherId(login, password) {
+    const { vaultClient, konnector } = this.props
+
+    const konnectorURI = get(konnector, 'vendor_link')
+    const konnectorName = get(konnector, 'name') || get(konnector, 'slug')
+
+    const cipherData = {
+      id: null,
+      type: 1,
+      name: konnectorName,
+      login: {
+        username: login,
+        password,
+        uris: konnectorURI ? [{ uri: konnectorURI, match: 0 }] : []
+      }
+    }
+
+    const cipher = await vaultClient.createNewCipher(cipherData, null)
+    await vaultClient.saveCipher(cipher)
+
+    return cipher.id
+  }
+
+  /**
+   * Find an existing cipher for the account and return its ID
+   *
+   * @param {string} login - the login to set in the cipher
+   * @param {string} password - the password to set in the cipher
+   *
+   * @returns {string} the cipher ID
+   */
+  async getExistingCipherIdForAccount(login, password) {
```

Comment:
Should we pass the `account` in params? and move this method to the account model? 

Also, we say `Find an existing cipher` but it seems that we create a `newCypher` 🤔

Reply:
Actually the `createCipher` seems not to be so well-named. It's more like a `createOrUpdate`